request/inspector.go

```diff
@@ -7,6 +7,17 @@ import (
 )
 
 type ResponseInspector interface {
+	// InspectResponse is passed a response to inspect.
+	//
+	// An inspector must not modify the response. Doing so could impact later
+	// inspectors.
+	//
+	// The state of the response's body is undefined. There could be multiple
+	// inspectors before or after any given inspector, so when reading the
+	// body, it's probably a good idea to restore it when done.
+	//
+	// Any error returned will simply be logged. This might be removed in the
```

Comment:
Why? It seems like you are changing the intention of the interface without a need.